core/include/webview/amalgamate_src_detail.hh

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#if defined(__cplusplus) && !defined(WEBVIEW_HEADER)
+
+#include "../../src/detail/engine_base.cc"
+
+/** TODO
+ *
+ * #include "../../src/detail/basic_result.cc"
+ * #include "../../src/detail/exceptions.cc"
+ * #include "../../src/detail/json.cc"
+ * #include "../../src/detail/native_library.cc"
+ * #include "../../src/detail/optional.cc"
+ * #include "../../src/detail/user_script.cc"
+ * 
+ * #include "../../src/detail/backends/cocoa_webkit.cc"
+ * #include "../../src/detail/backends/gtk_webkitgtk.cc"
+ * #include "../../src/detail/backends/win32_edge.cc"
+ * 
+ * #include "../../src/detail/platform/darwin/cocoa.cc"
+ * #include "../../src/detail/platform/darwin/objc.cc"
+ * #include "../../src/detail/platform/darwin/webkit.cc"
+ * 
+ * #include "../../src/detail/platform/linux/gtk/compat.cc"
+ * #include "../../src/detail/platform/linux/webkitgtk/compat.cc"
+ * #include "../../src/detail/platform/linux/webkitgtk/dmabuf.cc"
+ * 
+ * #include "../../src/detail/platform/windows/com_init_wrapper.cc
+ * #include "../../src/detail/platform/windows/dpi.cc
+ * #include "../../src/detail/platform/windows/dwmapi.cc
+ * #include "../../src/detail/platform/windows/iid.cc
+ * #include "../../src/detail/platform/windows/ntdll.cc
+ * #include "../../src/detail/platform/windows/reg_key.cc
+ * #include "../../src/detail/platform/windows/shcore.cc
+ * #include "../../src/detail/platform/windows/theme.cc
+ * #include "../../src/detail/platform/windows/user32.cc
+ * #include "../../src/detail/platform/windows/version.cc
+ * #include "../../src/detail/platform/windows/webview2/loader.cc
```

Comment:
These could be spread over time and accomplished by a variety of contributors.

The refactors should be fairly linear with low risk levels.